src/sys/unix.rs

```diff
@@ -1597,6 +1597,93 @@ impl crate::Socket {
         ))
         .map(|_| sbytes as usize)
     }
+
+    /// Set the value of the `TCP_USER_TIMEOUT` option on this socket.
+    ///
+    /// If set, this specifies the maximum amount of time that transmitted data may remain
+    /// unacknowledged or buffered data may remain untransmitted before TCP will forcibly close the
+    /// corresponding connection.
```

Comment:
Can you add that when `None` is used it will use the system default. That might be surprising behaviour.

Reply:
https://github.com/rust-lang/socket2/pull/241/commits/df14e0d8c5dd6ddc4056e0c4cad1509ea49993c8